### PR TITLE
Upgrade detekt

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: utils/src/main/kotlin/Utils.kt [hunk 1]
```diff
@@ -230,16 +229,13 @@ fun normalizeVcsUrl(vcsUrl: String): String {
     //     [scheme:][//authority][path][?query][#fragment]
     // where a server-based "authority" has the syntax
     //     [user-info@]host[:port]
-    val uri = try {
+    val uri = runCatching {
```
Comment (oheger-bosch) [inline utils/src/main/kotlin/Utils.kt:232]: A significant part of these issues seems to be related to the construction of possibly invalid URIs. Would it make sense to have a special utility function for this purpose, maybe something like this:
`fun <T> withUri(uriStr: String, defaultValue: T?, block: (URI) -> T): T?`
(The signature may need some fine-tuning.)

File: spdx-utils/src/main/kotlin/Extensions.kt [hunk 2]
```diff
@@ -32,7 +32,7 @@ import org.ossreviewtoolkit.spdx.SpdxExpression.Strictness
 /**
  * Return a string of hexadecimal digits representing the bytes in the array.
  */
-fun ByteArray.toHexString(): String = joinToString("") { String.format("%02x", it) }
+fun ByteArray.toHexString(): String = joinToString("") { String.format(null, "%02x", it) }
```
Comment (oheger-bosch) [inline spdx-utils/src/main/kotlin/Extensions.kt:35]: The `Locale` class has a special constant `ROOT`, which should be used instead of *null*.

Comment (sschuberth) [reply spdx-utils/src/main/kotlin/Extensions.kt:35]: Thanks, I wasn't aware of that constant.

File: utils/src/main/kotlin/Utils.kt (hunk 1 above)
Comment (sschuberth) [reply utils/src/main/kotlin/Utils.kt:232]: I was briefly thinking about something like this, too, but did not quickly find a proper signature for such a helper function. I'll probably do it as a follow-up.

**Review by oheger-bosch — APPROVED** [review]

File: utils/src/main/kotlin/Utils.kt (hunk 1 above)
Comment (sschuberth) [reply utils/src/main/kotlin/Utils.kt:232]: See https://github.com/oss-review-toolkit/ort/pull/3752.

**Pull request merged**